e2e/capacity_test.go

```diff
@@ -0,0 +1,211 @@
+
+		By("checking the pod is running that has possible scheduling PVC")
+		result, _, err := kubectl("get", "node", nodeName, "-o=json")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		var node corev1.Node
+		err = json.Unmarshal(result, &node)
+		Expect(err).ShouldNot(HaveOccurred())
+		size := func() string {
+			sizeStr, exists := node.Annotations["capacity.topolvm.cybozu.com/00default"]
+			if !exists {
+				Expect(errors.New("size is not found")).ShouldNot(HaveOccurred())
+			}
+			size, err := strconv.Atoi(sizeStr)
+			Expect(err).ShouldNot(HaveOccurred())
+			s := size >> 30
+			return strconv.Itoa(s)
+		}()
+
+		lvYaml := []byte(fmt.Sprintf(podPVCTemplateYAML, name, size, name, name, nodeName))
+		_, _, err = kubectlWithInput(lvYaml, "apply", "-f", "-")
+		cleanupList = append(cleanupList, lvYaml)
+		Expect(err).ShouldNot(HaveOccurred())
+		Eventually(func() error {
+			result, stderr, err := kubectl("get", "pvc", name, "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			var pvc corev1.PersistentVolumeClaim
+			err = json.Unmarshal(result, &pvc)
+			if err != nil {
+				return err
+			}
+			if pvc.Status.Phase != corev1.ClaimBound {
+				return errors.New("pvc status is not bound")
+			}
+
+			result, stderr, err = kubectl("get", "pods", name, "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			var pod corev1.Pod
+			err = json.Unmarshal(result, &pod)
+			if err != nil {
+				return err
+			}
+
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+					return nil
+				}
+			}
+
+			return errors.New("pod is not running")
+		}).Should(Succeed())
+
+		By("checking the pod is not running that has not possible scheduling PVC")
+		name2 := name + "2"
+
+		lvYaml2 := []byte(fmt.Sprintf(podPVCTemplateYAML, name2, size, name2, name2, nodeName))
+		_, _, err = kubectlWithInput(lvYaml2, "apply", "-f", "-")
+		cleanupList = append(cleanupList, lvYaml2)
+		Expect(err).ShouldNot(HaveOccurred())
+		Eventually(func() error {
+			result, stderr, err := kubectl("get", "pods", name2, "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			var pod corev1.Pod
+			err = json.Unmarshal(result, &pod)
+			if err != nil {
+				return err
+			}
+
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+					return errors.New("pod is running")
+				}
+			}
+
+			podInfo, stderr, err := kubectl("describe", "pods", name2)
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+			if !strings.Contains(string(podInfo), "did not have enough free storage") {
+				return errors.New("scheduling failed message is not found")
+			}
+
+			return nil
+		}).Should(Succeed())
+
+		if !isDaemonsetLvmdEnvSet() {
```

Comment:
Would you explain why this check is needed? This is not related to daetmonset lvmd, IMO.

Reply:
Sorry, I fixed it.
I want to check below things in this tests:
- the pod having a PVC that is able to schedule is running by the storage capacity tracking
- the pod having a PVC that is not able to schedule is not running by the storage capacity tracking because the target node doesn't have  enough storage
- when assigned a device class that has enough storage to the PVC, the pod having a PVC that is able to schedule is running by the storage capacity tracking even if the storage capacity of the default device class doesn't have enough storage